### ISSUE TITLE
Cleanup of Location, Place, Facility, Destination

### ISSUE THREAD

**Issue by knelson-farmbeltnorth** [ADAPT/Standard#45]
While there is some meaningful distinction between these entities, there are numerous issues and it is clear that they were not defined in a single, coherent vision.

 Location is defined as a Lat/Lon and its provenance.   
--ParentFacilityId shows up unexpectedly here with an explanation that needs more clarity.   When would someone use this workaround, and can we find another way?

Place is a Location extended with miscellaneous data
--There are miscellaneous GFF properties.   Per my comments at #38 , if we need the option to log at multiple levels of GFF hierarchy, then we should enforce only 1 level (Farm, Field or CropZone) is set.   The current enum provides a means for this.
--An optional DeviceElementID conceivably allows referencing Irrigation Equipment or some other static sensor.   Is linking to DeviceElement vs. DeviceElementConfiguration intentional?    
--An optional ContainerID conceivably allows referencing a bin, but there is no way to have hierarchical relationships (i.e., a bin site and its bins)
--In both of these cases, would it not make more sense for an IrrSystemConfiguration and Bin to reference their location vs. the other way around?

Facility seems to be in many ways a duplicate of Place, implemented at a different time, with a different set of type values in the RepresentationSystem than Place has in its Enum.   Facility allows for hierarchies, whereas Place currently doesn't.   It is missing a Location.    I'm assuming this was an oversight vs. some conscious design for persistent facilities that might move.

Destination seems completely redundant.   Unless we are tracking distance from source to destination, then a destination is a facility.

@aferreyra @strhea @dubnemo I know elements here have crept into the discussions around traceability and its associated entities, but I think we need to treat entities of physical location as their own problem set before we start confusing the discussion with transfer events.

At first glance, Place is a clumsy and confusing object.   I think fields, bins and center pivots should have locations vs. locations having them.   

The Location entity is clean enough, and if we extend Facility with a Location, then I don't think we need Place or Destination.   As I conceive of Facility (location + business purpose), it seems sufficiently flexible.     I'm sure there is lots more discussion to be had here, however.

**Comment by dubnemo** [comment]
@knelson-farmbeltnorth  
THIS is a loaded issue.  

There has been large debate around the definition of location and place during the 25+ years of my involvement when data exchange standards.  What I do know is 1) the location seems to be this more prevalent term in data exchange standards today, and 2) that location is not necessary fixed but relative, such as from an anchor point such as x, y, z from the corner of a semi-trailer where a grain probe takes a sample during grain receiving.  Place and location are close cousin synonyms and I don't recall anyone 'winning' the debate which term is the more 'abstract term'.  

The facility comment confuses me.  There are many more _contextualized definitions_ for that term than place or location, including 'an ability to do or learn something well and easily; a natural aptitude' ( french etymology).  Or even 'The definition of a facility is a building or room which was created to serve a specific purpose or is the ease of doing something.'

**Comment by knelson-farmbeltnorth** [comment]
@dubnemo I'm less concerned with what terms we use than the concepts themselves.    Facility indeed may have too many different connotations to be desirable.  I'm also less concerned with abstract theory than finding a pragmatic model of physical place that works for agribusiness application.

We have a jumble of concepts here.   Geodetic coordinates, where we got them, their offset from something else, the functional disposition of whatever is located at the coordinates, hierarchical relationships that may exist at this location, etc.    We don't need to go back to first principles, but I don't think what we have in the model today works together correctly.

**Comment by dubnemo** [comment]
@knelson-farmbeltnorth perhaps the next step is to look at the original UML model to better understand the intent, and if the implementation meets the intended goals, or if the model itself has a flaw based on real-world experiences.

**Comment by knelson-farmbeltnorth** [comment]
Re-examining this, we can simplify these objects to simply Location and Facility with the following definitions:
Location: A physical point, defined by a latitude and a longitude.
Facility: A physical site used by an Entity.   A Facility can be a building site, bin site, field installation, etc.

Destination was never fully implemented.  A reference to it exists only on Load, and since there is no place to store Destinations, it has never been used.   We can simply change Load Destination Id to Load Desination Facility Id to accomplish what was desired.

There is a comment on Location.ParentFacilityId:  "Locations in PAIL have Ids; they don't in ADAPT. This provides a way to enable using ADAPT facilities to stand in for Locations (by reference) when needed."    If a location has a persistent Id, it is really a facility.   A Location in ADAPT is just coordinates and their provenance.   As such, Parent Facility Id should be removed.

Place came in with the PAIL/Observations changes.  It is an abstract representation of any type of place, accounting for fields, containers, irrigation devices, or facilities.    As I've defined Facility above, it accomplishes all of these purposes except Field.     It gets muddy trying to merge Field into Facility since Field is such a primary object within ADAPT.   Where needed, we should include an optional FieldId next to FacilityId.

**Comment by knelson-farmbeltnorth** [comment]
For clarity, @aferreyra  has suggested Location be named Position for consistency with other standards.   I endorse this.

**Comment by knelson-farmbeltnorth** [comment]
Changes agreed at Oct 26, 2022 meeting
